vertical-pod-autoscaler/recommender/base/circularbuffer.go

```diff
@@ -0,0 +1,62 @@
+package base
+
+// A queue with a fixed maximum capacity. Elements are inserted/removed in the
+// FIFO order. Elements are removed from the buffer when it runs out of capacity
+// and a new element is inserted.
+type CircularBuffer interface {
```

Comment:
The comments should start with function/struct/interface name. Here and elsewhere.